src/dialect/mssql/mssql-driver.ts

```diff
@@ -56,7 +56,10 @@ export class MssqlDriver implements Driver {
       },
       // @ts-ignore `tarn` accepts a function that returns a promise here, but
       // the types are not aligned and it type errors.
-      validate: (connection) => connection.validate(),
+      validate:
+        this.#config.tarn.options.validateConnections === false
+          ? () => true
+          : (connection) => connection.validate(),
```

Comment:
`tarn`'s default is `() => true`. https://github.com/Vincit/tarn.js/blob/master/src/Pool.ts#L132

```suggestion
        this.#config.tarn.options.validateConnections === false
          ? undefined
          : (connection) => connection.validate(),
```

Reply:
Making this change would require updating the [`validate` type](https://github.com/tmrclark/kysely/blob/mssql-pooling-updates/src/dialect/mssql/mssql-dialect-config.ts/#L202). We can also update it to remove the need for the `@ts-ignore` if we change it to `validate?: (resource: R) => boolean | Promise<boolean>`